src/grammar/query.rs

```diff
@@ -206,6 +210,69 @@ pub(crate) fn parse_order_by_clause(p: &mut Parser) {
     p.finish();
 }
 
+pub(crate) fn parse_group_by_clause(p: &mut Parser) {
+    p.start(SyntaxKind::GroupByClause);
+    p.expect(T![group]);
+    p.expect(T![by]);
+    safe_loop!(p, {
+        if [T![rollup], T![cube]].contains(&p.current()) {
+            parse_rollup_cube_clause(p);
+        } else if p.at(T![grouping]) {
+            parse_grouping_sets_clause(p);
+        } else {
+            parse_expr(p);
+        }
```

Comment:
Lets do this with a `match` over `p.current()`, should be a bit easier to read :smile: